doc/feature/compartmentReport.md

```diff
@@ -0,0 +1,422 @@
+at the reader, because it can be created locally based on the selected cell
+subset regardless of which cells are being published by the writer.
+That said, soma only reports may receive special treatment for efficiency
+reasons, but that should probably also imply an API extension.
+
+The reason for sending the mapping as part of every (per cell) frame is also
+simplicity. If the mapping is a separate piece of information that needs to
+be requested by clients, it adds complexity in several points. The first one
+is the constructor, that instead of simply creating a subscriber, needs to
+request the mapping synchronously. Changing the cell set of interest on the
+reader is also more complicated, since it requires again a request that with
+the current API must be synchronous. The proposed approach also avoids having
+to add a finishMapping function to know then the mapping is complete and can
+be sent to receivers potentially waiting in their constructors.
+Related to the issue above we could distinguish between receiving a global
+mapping or per cell mappings. If the mapping was requested separately, having
+a whole frame, makes all the code more complicated. The overhead of including
+the per cell mapping inside the data message can be estimated knowing the
+maximum and average numbers of comparments section (unreported sections need
+to be included in this calculation). For example, assuming at most 16
+compartments per section and 4 compartments on average, the overhead is an
+additional 1/32th of the frame size. In this case, for every 4 float numbers
+(the compartment data) we have to add a compartment count that can be enconded
+with 4 bits, in other words we have 4/(4*32) extra bits for each data bit. The
+actual overhead still needs to be figured out, but anything around this 3%
+seems reasonable, overall considering that is data size is a problem a big
+reduction can be achived using compression of the floating point data.
+
+The messages of the protocol are:
+ - Publiser to subscribers:
+   * FRAME_DATA
+   * FRAME_BARRIER
+   * REPORT_METADATA
+ - Subscribers to publisher:
+   * REQUEST_METADATA
+
+The fields of a frame data message are the following:
+ - cell GID
+ - timestamp
+ - compartment data (n floats)
+ - compartment counts (m integers packed as 4 or 8 bit n = Sum counts)
+
+The timestamp is guaranteed to be the same for all cells in the same frame which
+means that a frame can be considered finished when the last received timestamp
+is different to the previous one. Nevertheless, in order to avoid receivers wait
+until the first message of the next frame arrives, the FRAME_BARRIER message
+is used to signal the end of a frame (this message is an empty event). The
+FRAME_BARRIER message is used before the very first frame data message, this
+is done because otherwise receivers do not have enough information to know
+if the very first frame is a complete frame or they joined late.
```

Comment:
It seems redundant, but may also be added robustness. In Collage I've seen weird cases where packets arrive late, so being able to correlate a barrier to a frame explicitly instead of implicitly is imo a good idea.